specs/interop/liquidity_migration.md

```diff
@@ -0,0 +1,358 @@
+# Liquidity Migration
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Overview](#overview)
+- [Implementation changes](#implementation-changes)
+  - [External functions](#external-functions)
+    - [`convertToSuper`](#converttosuper)
+    - [`convertFromSuper`](#convertfromsuper)
+  - [Events](#events)
+    - [`ConvertedToSuper`](#convertedtosuper)
+    - [`ConvertedFromSuper`](#convertedfromsuper)
+  - [Internal Functions](#internal-functions)
+    - [`checkPair`](#checkpair)
+- [Diagram](#diagram)
+- [Invariants](#invariants)
+- [Considerations](#considerations)
+- [Appendix](#appendix)
+  - [Access Control](#access-control)
+  - [Decimal management](#decimal-management)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `OptimismMintableERC20` tokens and `L2StandardToken`s (_legacy tokens_),
+which correspond to locked liquidity in L1, are incompatible with interop.
+Legacy token owners must convert into a `SuperchainERC20` representation, as defined in the [standard](token-bridging.md),
+to move across the Superchain.
+
+The conversion method will use the `L2StandardBridge` mint/burn rights
+over the legacy tokens to allow easy migration to and from the
+corresponding `SuperchainERC20`.
+
+## Implementation changes
+
+The `L2StandardBridge` will be extended to include the following functions and events:
+
+### External functions
+
+#### `convertToSuper`
+
+Converts `_amount` of legacy tokens with address `_legacyAddr`
+to the same `_amount` of `SuperchainERC20` tokens in address `_superAddr`.
+
+```solidity
+convertToSuper(address _legacyAddr, address _superAddr, uint256 _amount)
+```
+
+The function will
+
+1. Check that `_legacyAddr` and `_superAddr` are valid and paired.
+2. Check both tokens have the same amount of decimals.
```

Comment:
I think this is probably the best option. Figuring out how to handle decimal changes is a big can of worms.

Reply:
Yes, that seems to be the consensus. 

For context, L1 deposits do not check decimals at all, and it's perfectly the same to do the same here (remove the check altogether). In that case, the market should adjust the pricing for each representation.
So, both approaches are safe:
- enforce the same decimals
- don't enforce and let the market price the assets
I think it's mostly a UX decision.